src/Util.h

```diff
@@ -1572,59 +1572,71 @@ class LRUCache<T, 0> {
 //                           Hint / Cache Profiling
 // -------------------------------------------------------------------------------
 
-/**
- * A utility function to determine whether hints-profiling is enabled or
- * disabled;
- */
-inline bool isHintsProfilingEnabled() {
-    return std::getenv("SOUFFLE_PROFILE_HINTS") != nullptr;
-}
-
 /**
  * A utility class to keep track of cache hits/misses.
  */
 class CacheAccessCounter {
-    bool active;
+#ifdef _SOUFFLE_STATS
     std::atomic<std::size_t> hits;
     std::atomic<std::size_t> misses;
+#endif
 
 public:
-    CacheAccessCounter(bool active = isHintsProfilingEnabled()) : active(active), hits(0), misses(0) {}
+    CacheAccessCounter()
+#ifdef _SOUFFLE_STATS
```

Comment:
Can this be `= default;` for both cases, since you're initialising to 0 anyway. In which case no `#ifdef` is needed.

Reply:
Actually, can either the whole class be wrapped in a single ifdef with a complete class in each branch, or could there be no ifdefs and the complete class is present - but will go unused. The overhead for the latter is 2x4 extra atomics in BTree that are present, but not used